File: translations/codenet/uncommented/C2Go/gpt-4o/Code/s190193281.go
package main

import (
	"fmt"
)

const N = 1000000
const Q = 50000

var n int
var S [N]int
var T [Q]int

func binarySearch(key int) int {
	left, right := 0, n
	for left < right {
		mid := (left + right) / 2
		if S[mid] == key {
			return 1
		} else if key < S[mid] {
			right = mid
		} else {
			left = mid + 1
		}
	}
	return 0
}

func main() {
	var q, c int

	fmt.Scan(&n)
	for i := 0; i < n; i++ {
		fmt.Scan(&S[i])
	}

	fmt.Scan(&q)
	for i := 0; i < q; i++ {
		fmt.Scan(&T[i])
		if binarySearch(T[i]) == 1 {
			c++
		}
	}

	fmt.Println(c)
}

// <END-OF-CODE>

File: translations/codenet/uncommented/C2C++/gpt-4o/Code/s190193281.cpp
#include <iostream>
#include <vector>
#include <algorithm>

#define N 1000000
#define Q 50000

int n;
std::vector<int> S(N);
std::vector<int> T(Q);

int binarySearch(int key) {
    int left = 0, right = n, mid;
    while (left < right) {
        mid = (left + right) / 2;
        if (S[mid] == key) return 1;
        else if (key < S[mid]) right = mid;
        else left = mid + 1;
    }
    return 0;
}

int main() {
    int q, c = 0;

    std::cin >> n;
    for (int i = 0; i < n; i++) {
        std::cin >> S[i];
    }

    std::cin >> q;
    for (int i = 0; i < q; i++) {
        std::cin >> T[i];
        if (binarySearch(T[i]) == 1) c++;
    }

    std::cout << c << std::endl;

    return 0;
}

// <END-OF-CODE>
